stdlib/public/core/ArrayBuffer.swift

```diff
@@ -567,6 +571,78 @@ extension _ArrayBuffer {
     }
   }
 
+  @inlinable @_alwaysEmitIntoClient
+  static var associationKey: UnsafeRawPointer {
+    //We never dereference this, we just need an address to use as a unique key
+    UnsafeRawPointer(Builtin.addressof(&_swiftEmptyArrayStorage))
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func getAssociatedBuffer() -> _ContiguousArrayBuffer<Element>? {
+    let getter = objc_getAssociatedObject as @convention(c)(
+      Any,
+      UnsafeRawPointer
+    ) -> Any?
+    let typedGetter = unsafeBitCast(
+      getter,
+      to: (@convention(c)(
+        AnyObject,
+        UnsafeRawPointer
+      ) -> AnyObject?).self
+    )
+    if let assoc = typedGetter(
+      _storage.objCInstance,
+      _ArrayBuffer.associationKey
+    ) {
+      let buffer = unsafeBitCast(
+        assoc,
+        to: _ContiguousArrayStorage<Element>.self
+      )
+      return _ContiguousArrayBuffer(buffer)
+    }
+    return nil
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func setAssociatedBuffer(_ buffer: _ContiguousArrayBuffer<Element>) {
+    let setter = unsafeBitCast(getSetAssociatedObjectPtr(), to: (@convention(c)(
```

Comment:
This ridiculous dance is a way of tricking swiftc into calling objc_setAssociatedObject() directly without a) being able to import the definition, or b) going through the importer's id -> Any thunking